internal/scheduler/scheduler.go

```diff
@@ -485,18 +524,68 @@ func (s *Scheduler) generateUpdateMessagesFromJob(job *jobdb.Job, jobRunErrors m
 			}
 			events = append(events, jobSucceeded)
 		} else if lastRun.Failed() {
-			job = job.WithFailed(true).WithQueued(false)
-			runError := jobRunErrors[lastRun.Id()]
-			jobErrors := &armadaevents.EventSequence_Event{
-				Created: s.now(),
-				Event: &armadaevents.EventSequence_Event_JobErrors{
-					JobErrors: &armadaevents.JobErrors{
-						JobId:  jobId,
-						Errors: []*armadaevents.Error{runError},
+			requeueJob := lastRun.Returned() && job.NumAttempts() < s.maxAttemptedRuns
+
+			if requeueJob && lastRun.RunAttempted() {
```

Comment:
When we become leader (either on startup or through failover) we pass all jobs to this function.  In this case we will get jobs that have already been requeued fall into this code branch.  In that case I *think* the code will do the right thing- i.e. it'll end up with the jobs queued with the correct anti affinity set but it might be either writing a test to be sure, or seeing if we can already detect the job is in the correct state and short circuit.

Reply:
I'm relatively confident it will be sufficient to check it is queued - however we could also validate the affinity is as we expect

However this would only handle if we managed to get into an unexpected state - but maybe that isn't a bad thing